docs/how-to/troubleshoot_deployment_issues.md

```diff
@@ -0,0 +1,199 @@
+# Troubleshoot the deployment issues
+
+This guide provides step-by-step troubleshooting actions to remediate deployment issues. We hope you don't need this guide. If you encounter an issue and aren't able to address it via this guide, please raise an issue [here][Bug Report].
+
+## 1. Terraform failed to deploy the Charmed Aether SD-Core because of configuration mismatch
+
+### Symptoms
+
+Terraform may fail to deploy the Charmed Aether SD-Core modules. When the `terraform apply -auto-approve` command is run, the deployment fails with the wrong integration endpoints or the wrong object attributes etc. as following:
+
+```console
+$ terraform apply -auto-approve 
+╷
+│ Error: Unsupported attribute
+│ 
+│   on main.tf line 54, in resource "juju_integration" "cu-nms":
+│   54:     endpoint = module.sdcore.fiveg_identity_endpoint
+│     ├────────────────
+│     │ module.sdcore is a object
+│ 
+│ This object does not have an attribute named "fiveg_identity_endpoint".
+```
+
+### Recommended Actions
+
+[Validate the Terraform modules][Validate Terraform Configuration]:
+
+```shell
+$ terraform validate
+Success! The configuration is valid.
+```
+
+The command should not fail. If it does, please open a [bug report][Bug Report].
```

Comment:
I'm not sure if we need this whole point. If the deployment fails, the user will open a bug anyway and telling them to run `terraform validate` just to see that it fails doesn't seem to bring a lot of value.
WDYT?

Reply:
There can be some cases that `terraform apply -auto-approve` fails but `terraform validate` is successful. This indicates another problem. The user will create a bug in both cases but the validation step provides more insight.

Reply:
This case is removed.